src/routers/SwitchRouter.js

```diff
@@ -226,7 +263,7 @@ export default (routeConfigs, config = {}) => {
               routes,
               index: activeChildIndex,
             };
-            return getNextState(prevState, nextState);
+            return updateRouteKeyHistory(getNextState(prevState, nextState));
```

Comment:
can we move `updateRouteKeyHistory` into `getNextState`? afraid of it being too easy to accidentally not update the route key history in the future

Reply:
done